x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/hooks.test.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+const mockAction = jest.fn();
+
+let mockSubcription: (state: object) => void;
+const mockDriver = {
+  state: { foo: 'foo' },
+  actions: { bar: mockAction },
+  subscribeToStateChanges: jest.fn().mockImplementation((fn) => {
+    mockSubcription = fn;
+  }),
+  unsubscribeToStateChanges: jest.fn(),
+};
+
+jest.mock('react', () => ({
+  ...(jest.requireActual('react') as object),
+  useContext: jest.fn(() => ({
+    driver: mockDriver,
+  })),
+}));
+
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { mount } from 'enzyme';
+
+import { useSearchContextState, useSearchContextActions } from './hooks';
+
+describe('hooks', () => {
+  describe('useSearchContextState', () => {
+    it('exposes search state', () => {
+      const TestComponent = () => {
+        const { foo } = useSearchContextState();
+        return <div>{foo}</div>;
+      };
+
+      const wrapper = mount(<TestComponent />);
+      expect(wrapper.text()).toEqual('foo');
+    });
+
+    it('subscribes to state changes', () => {
+      const TestComponent = () => {
+        const { foo } = useSearchContextState();
+        return <div>{foo}</div>;
+      };
+
+      const wrapper = mount(<TestComponent />);
+
+      act(() => {
+        mockSubcription({ foo: 'bar' });
+      });
+
+      expect(wrapper.text()).toEqual('bar');
+    });
+
+    it('unsubscribes to state changes when unmounted', () => {
+      const TestComponent = () => {
+        const { foo } = useSearchContextState();
+        return <div>{foo}</div>;
+      };
```

Comment:
No major concerns. I've done that before. A consideration though is that makes adding future tests just a bit clunkier to add in the future, and a bit more error prone since things aren't reset in between tests.

Reply:
Right yeah, it definitely only works for test blocks/components that you know are going remain static and don't need new props etc. between assertions. Just thought it applied here in this situation

Reply:
57673cef5f0cc46028f778f8f0d9a9b3b44b82d4